oa-enterprise/lib/omniauth/strategies/cas.rb

```diff
@@ -8,6 +8,29 @@ class CAS
       autoload :Configuration, 'omniauth/strategies/cas/configuration'
       autoload :ServiceTicketValidator, 'omniauth/strategies/cas/service_ticket_validator'
 
+      attr_accessor :raw_info
+      alias_method :user_info, :raw_info
+      
+      option :uid_key,              'user'
+ 
+       # As required by https://github.com/intridea/omniauth/wiki/Auth-Hash-Schema
+       AuthHashSchemaKeys = %w{name email first_name last_name location image phone}
+       info do
+         prune!({
+           :name       => raw_info['name'],
+           :email      => raw_info['email'],
+           :first_name => raw_info['first_name'],
+           :last_name  => raw_info['last_name'],
+           :location   => raw_info['location'],
+           :image      => raw_info['image'],
+           :phone      => raw_info['phone']
+         })
+       end
+       
+      uid do
```

Comment:
uid is only available in [modern implementations of `OmniAuth::Strategy`](https://github.com/omniauth/omniauth/blob/master/lib/omniauth/strategy.rb#L316)